### ISSUE TITLE
Configs: add a config that turns off all type-aware linting

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#6215]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22preset+config+change%22+) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Description

Our current guidance on how to setup an ESLint config is to turn everything on in the root of the config and be done with it. This advice includes turning on type-aware linting on all files.

However for a lot of codebases this can create a crappy experience because there are a lot of files NOT included in a tsconfig that ESLint will run over. This in turn causes the dreaded "The file must be included in at least one of the projects provided" error.

There's not really a good way around this right now, sadly. [Our advice](https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file) is pretty long winded and not the easiest thing to put into practice, especially if you don't actually want to lint the file with type-info. Possible workarounds include:
1) `.eslintignore` the files.
2) create a `tsconfig.eslint.json` that includes the files.
3) manually turn off type-aware rules AND `parserOptions.project` on the files.
4) rework your config so that you only enable type-aware linting on specific files using `overrides` config.

All of those options suck for a user because:
1) these files are often files that you want to do at least basic lint checks on
2) these files are often `.js` files with poor-to-no types or are config files where most type-aware rules aren't going to catch anything useful (if anything at all)
3) it's difficult to know exactly what type-aware rules you have turned on due to config composition
4) it's hard to rework your entire config without accidentally missing files that should be type-aware linted.

I realised that we're in a good position to help people automate this. We know what rules need type-info - so why not generate a config for it? That way the solution to the problem can simply be "use overrides to disable type-aware linting by extending this config", i.e.
```js
module.exports = {
  // ...
  overrides: [
    // disable type-aware linting for some files
    {
      files: ['.eslintrc.js'/*, ... */],
      extends: ['plugin:@typescript-eslint/disable-type-aware-linting'],
    },
  ],
}
```

This would be a substantially better UX and is a pretty clean solution!

**Comment by Josh-Cena** [comment]
Yesss...! This would make https://twitter.com/SidaChen63/status/1581749782980153344 so much easier

**Comment by Josh-Cena** [comment]
Closed by https://github.com/typescript-eslint/typescript-eslint/pull/6470; idk why the keyword isn't working.

**Comment by bradzacher** [comment]
the keyword only works if you merge into the default branch sadly
